Add gelu gradient for pytorch

xiaomengy opened the pull request:
Summary: Add gelu gradient for pytorch

Differential Revision: D15589816

soumith commented on test/test_nn.py:
```diff
@@ -6067,16 +6067,19 @@ def _gelu_ref(X):
-            ref = _gelu_ref(X.numpy())
+            ref = _gelu_ref(X.detach().numpy())
             self.assertEqual(res, ref)
+            gradcheck(F.gelu, [X], eps=1e-4, atol=1e-3)
```
this looks suspicious. gradchecks happen in double precision, so a tolerance of 1e-3 looks really high and a custom eps is usually not needed. Any idea what is going on? Can you check some sample inputs to inspect.

BIT-silence replied:
This part just came from testing the numerical stability for gradchecks. Atol has been removed and now it is using default value.

soumith reviewed (approved):
please check inline comment and provide resolution on what's up with the tolerance adjustment. Once you get clarity into that, and verify it's not a bug, do land.

Things reviewed in the diff:
- MKL and non-MKL implementations match in formula
- CUDA and CPU implementation match in formula

Things not reviewed in the diff:
- gradient formula is correct (relying on gradcheck to say it's right)

xiaomengy committed:
Add gelu activation in pytorch

Summary: Add gelu activation forward on CPU in pytorch

Differential Revision: D15400974

fbshipit-source-id: 1c59104bea69cbe26ab96921e242131890db657e

xiaomengy committed:
Add gelu gradient for pytorch (#21237)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/21237

Add gelu gradient for pytorch

Reviewed By: zheng-xq

Differential Revision: D15589816

fbshipit-source-id: 2feb4ed779cda1dec3fe03fcfba29861b4a86d12
